encoders/firebase-decoders-json/src/main/java/com/google/firebase/decoders/AnnotatedFieldHandler.java

```diff
@@ -18,7 +18,21 @@
 import androidx.annotation.Nullable;
 import java.lang.annotation.Annotation;
 
-public interface FieldModifier<U extends Annotation> {
+/**
+ * {@link AnnotatedFieldHandler} provide a way to handle field values based on its annotation.
+ *
+ * <p>Each annotation type should be registered along with associated {@link AnnotatedFieldHandler},
+ * non-registered annotation will be skipped in the process of decoding.
```

Comment:
Worth also noting that annotations themselves need to be annotated with `@ExtraProperty`